mockall/examples/ffi.rs

```diff
@@ -1,35 +1,39 @@
 // vim: tw=80
 //! An example of unit testing involving mocked FFI functions
 
-#![cfg(unix)]
-
+use std::ptr;
 #[cfg(test)]
 use mockall::automock;
 use mockall_double::double;
 
 #[cfg_attr(test, automock)]
 pub mod mockable_ffi {
     extern "C" {
-        pub fn getuid() -> u32;
+        pub fn time(timer: *const u64) -> u64;
     }
 }
 
 #[double]
 use mockable_ffi as ffi;
 
-fn getuid() -> u32 {
-    unsafe { ffi::getuid() }
+fn time() -> u64 {
+    unsafe { ffi::time(ptr::null()) }
 }
 
 fn main() {
-    println!("My uid is {}", getuid() );
+    println!("Seconds since epoch: {}", time() );
 }
 
 
 #[test]
-fn getuid_test() {
-    let ctx = ffi::getuid_context();
+fn time_test() {
+    let seconds_from_epoch = std::time::SystemTime::now()
```

Comment:
The example would be both clearer, and more distinct from regular behavior, if you simply return a fixed constant.